### PR TITLE
Add additional istiod keepalive arguments

### PR DESCRIPTION
**Pull request by johndlong**
Expose additional arguments in istiod around keepalive interval/timeouts.  The [flags](https://github.com/istio/istio/blob/master/pkg/keepalive/options.go#L64) are already built into istiod, but only available to customize using advanced techniques like a [custom install profile](https://gist.github.com/johndlong/b2246d29344d10a368772c966c101ba0) or helm post-renderer.  Helm post-renderer in particular is less than ideal since it needs to be executed on subsequent upgrades in addition to the install.  It would be preferred to be consistent with the `keepaliveMaxServerConnectionAge` arg that is currently exposed.

**To help us figure out who should review this PR, please put an X in all the areas that this PR affects.**

- [ ] Ambient
- [ ] Configuration Infrastructure
- [ ] Docs
- [X] Installation
- [ ] Networking
- [ ] Performance and Scalability
- [ ] Policies and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure

**Please check any characteristics that apply to this pull request.**

- [ ] Does not have any [user-facing](https://github.com/istio/istio/tree/master/releasenotes#when-to-add-release-notes) changes. This may include CLI changes, API changes, behavior changes, performance improvements, etc.

### CODE REVIEW

File: manifests/charts/istio-control/istio-discovery/values.yaml [hunk 1]
```diff
@@ -53,6 +53,13 @@ pilot:
 
   plugins: []
 
+  # The time interval if no activity on the connection it pings the peer to see if the transport is alive
+  keepaliveInterval: 30s
+
+  # After having pinged for keepalive check, the client/server waits for a duration of keepaliveTimeout
+  # and if no activity is seen even after that the connection is closed.
+  keepaliveTimeout: 10s
+
```
Comment (jacob-delgado) [inline manifests/charts/istio-control/istio-discovery/values.yaml:62]: Given the if conditional above, I'd prefer we either (1) comment these out or (2) remove them. I'm fine with them as hidden values in the helm chart.

Comment (johndlong) [reply manifests/charts/istio-control/istio-discovery/values.yaml]: Started going with 2, but then a `make gen` removes references from `gen-istio.yaml` and started feeling awkward compared to having it explicitly defined in the operator package.

File: releasenotes/notes/42398.yaml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+
+apiVersion: release-notes/v2
+kind: feature
+area: installation
+issue:
+  - https://github.com/istio/istio/pull/42398
```
Comment (zhlsunshine) [inline releasenotes/notes/42398.yaml:6]: This should be an issue which report the bug handled by this PR

**Comment by johndlong** [comment]
Created a feature request per @zhlsunshine's comment, now this PR resolves #43256

**Review by hzxuzhonghu — APPROVED** [review]
@howardjohn to confirm if this is not complete changing all helm charts

**Comment by howardjohn** [comment]
see https://github.com/istio/istio/pull/43166#issuecomment-1426249306

File: manifests/charts/istio-control/istio-discovery/files/gen-istio.yaml [hunk 3]
```diff
@@ -1691,6 +1691,8 @@ spec:
           - cluster.local
           - --keepaliveMaxServerConnectionAge
           - "30m"
+          - --keepaliveInterval=30s
```
Comment (costinm) [inline manifests/charts/istio-control/istio-discovery/files/gen-istio.yaml:1694]: Mesh config please ? Or env.

Comment (hzxuzhonghu) [reply manifests/charts/istio-control/istio-discovery/files/gen-istio.yaml:1694]: It was already in the flag

Comment (hzxuzhonghu) [reply manifests/charts/istio-control/istio-discovery/files/gen-istio.yaml:1694]: I think this pr is completing the charts

**Review by jacob-delgado — APPROVED** [review]

**Pull request merged**